openwisp_utils/admin_theme/theme.py

```diff
@@ -0,0 +1,66 @@
+from django.core.exceptions import ImproperlyConfigured
+
+from . import settings as app_settings
+
+THEME_LINKS = app_settings.OPENWISP_ADMIN_THEME_LINKS.copy()
+THEME_JS = app_settings.OPENWISP_ADMIN_THEME_JS.copy()
+
+
+def register_theme_link(links):
+    if not isinstance(links, list):
+        raise ImproperlyConfigured(
+            '"openwisp_utils.admin_theme.theme.register_theme_link"'
+            ' accepts "list" of links.'
+        )
+    # We don't raise ImproperlyConfigured exception if
+    # link is already present in THEME_LINKS because
+    # user might specify a link in project settings which is
+    # already registered by an application.
+    # This would lead to time-consuming debugging for little returns.
+    for link in links:
+        if link not in THEME_LINKS:
+            THEME_LINKS.append(link)
+
+
+def unregister_theme_link(links):
+    if not isinstance(links, list):
+        raise ImproperlyConfigured(
+            '"openwisp_utils.admin_theme.theme.unregister_theme_link"'
+            ' accepts "list" of links.'
+        )
+    for link in links:
+        try:
+            THEME_LINKS.remove(link)
+        except ValueError:
+            raise ImproperlyConfigured(
+                f'{link["href"]} was not added to OPENWISP_ADMIN_THEME_LINKS'
+            )
```

Comment:
This is good but also inconsistent with what's being done in the register function: the register won't complain if there's a duplicate but the unregister will complain if there's nothing to remove, the behavior is not consistent even within the same file. We have to avoid adding inconsistencies across the code base, let's decide one way and stick to it. I believe that failing early and loud if a piece of code is badly written (tries to register an item which has been already registered or tries to remove an item which is not present) is better because it allows developers to find out mistakes early instead of later on when they least expect it.